### PR TITLE
Service/Client Split

### PR DESCRIPTION
**Pull request by garvankeeley**
ggp and I have been working on a service/client split, as required to create a service-only library for fennec.

### CODE REVIEW

File: src/org/mozilla/mozstumbler/client/mapview/MapActivity.java [hunk 1]
```diff
@@ -18,21 +18,20 @@
...
 import java.util.Collections;
 import java.util.List;
-
 import org.json.JSONArray;
 import org.json.JSONException;
 import org.json.JSONObject;
-
-import org.mozilla.mozstumbler.cellscanner.CellInfo;
-import org.mozilla.mozstumbler.cellscanner.CellScanner;
-import org.mozilla.mozstumbler.communicator.Searcher;
```
Comment (tamcap) [inline src/org/mozilla/mozstumbler/client/mapview/MapActivity.java:35]: Fix required. Searcher not in scope anymore.

**Comment by tamcap** [comment]
There wasn't much time for reviews. I found at least one critical bug (see my comment to MapActivity.java).

File: src/org/mozilla/mozstumbler/client/mapview/MapActivity.java (hunk 1 above)
Comment (garvankeeley) [inline src/org/mozilla/mozstumbler/client/mapview/MapActivity.java:35]: Package scoped.

**Comment by garvankeeley** [comment]
Sorry about that, this is a branch ggp and I have been working off for some time, and some of the design was discussed with cpeterson. So it is peer-reviewed in that sense. It is too many code changes to leave in the PR state with no one officially assigned to review it. I have been running this version of stumbler for a while, it should match the previous version's functionality.
Hands-on reviews of the changes are good to have too, so I can provide pre-release builds if there is an interest.

File: src/org/mozilla/mozstumbler/client/mapview/MapActivity.java (hunk 1 above)
Comment (tamcap) [inline src/org/mozilla/mozstumbler/client/mapview/MapActivity.java:35]: Ohhh, I see :+1:

**Comment by tamcap** [comment]
No worries. Apparently the bug didn't really exist. I will make a local build and give it a run around too.

**Pull request merged**